### ISSUE TITLE
Global coffin inconsistent local/global assignment

### ISSUE THREAD

**Issue by blefloch** [latex3/latex3#1443]
As it turns out, `\vcoffin_gset:Nnn` shares an auxiliary with the local version, and both use under the hood `\prop_put:Nnn` to set coffin poles.  This trips up the l3debug check-declarations code, which produces the (correct) error `Inconsistent local/global assignment`.  Presumably the poles are then lost at the end of the group, but I didn't track that carefully.
```
\documentclass{article}
\ExplSyntaxOn
\debug_on:n { check-declarations }
\ExplSyntaxOff
\begin{document}
\ExplSyntaxOn
\vcoffin_gset:Nnn \g_tmpa_coffin { 10pt } { XXX } % gives the error
\coffin_typeset:Nnnnn \g_tmpa_coffin { t } { l } { 0pt } { 0pt }
\ExplSyntaxOff
\end{document}
```

**Comment by josephwright** [comment]
Isn't this addressed by 08347a04bfb9e4ecf08c3356c4dce02143589b25?

**Comment by blefloch** [comment]
Yes, indeed, thanks!  I came across the problem when working on my #1059 pull request, in a branch that is 30 commits behind main.
